### ISSUE TITLE
"serializer-test" generates integration test in unit test folder

### ISSUE THREAD

**Issue by Turbo87** [emberjs/data#4267]
I have a serializer that is mainly used for normalization of API responses. The default "serializer-test" however is focused on serializing model records and is actually more of an integration test since it's actually testing the `serialize()` method on the model.

I'm not sure what the best way is to improve the situation here, but I thing that this could probably be improved to be a bit more intuitive.

**Comment by Turbo87** [comment]
just noticed that there is also an issue with generating an `application` serializer-test, because it depends on `model:application` which obviously doesn't exist.

**Comment by Turbo87** [comment]
@bmac @pangratz any comments?

**Comment by pangratz** [comment]
I agree that the generated test for the serializer could definitely be better and the current one could actually be seen as kind of an integration test. I think there are cases though where unit testing a serializer are ok; take testing a custom implementation for `keyForAttribute` for example...

I think the current burden with writing serializer / adapter tests is that there are not really test helpers which allow you to quickly setup an environment where a specific case can be tested. This might get better once #3742 is tackled 😔

**Comment by pangratz** [comment]
FYI there is a ticket for the "no model for application" issue, tracked in #4436.
